src/components/Comment.jsx

```diff
@@ -9,9 +12,16 @@ const Comment = ({ comment }) => {
       <div className="comment-right-part">
         <div className="comment-info">
           <div className="comment-author">{comment.author}</div>
-          <div>{Date(comment.created_at)} </div>
+          <div>{new Date(comment.created_at).toUTCString()} </div>
         </div>
         <div className="comment-body">{comment.body}</div>
+        <div className="comment-buttons">
+          {canDelete && (
+            <div className="comment-delete" onClick={deleteComment}>
```

Comment:
You're not passing the comment_id to the deleteComment function here which is why it can't find which one to delete and gets undefined.

the onClick expects a function, and you can't pass a function invocation (`deleteComment(commentId)`)

So we have to pass an inline function like this `onClick= {( ) => deleteComment(comment.comment_id)}`